lib/nxdk/Makefile

```diff
@@ -0,0 +1,33 @@
+NXDK_SRCS := \
+	$(NXDK_DIR)/lib/nxdk/mount.c \
+	$(NXDK_DIR)/lib/nxdk/path.c
+
+NXDK_OBJS = $(addsuffix .obj, $(basename $(NXDK_SRCS)))
+
+$(NXDK_DIR)/lib/libnxdk.lib: $(NXDK_OBJS)
+
+main.exe: $(NXDK_DIR)/lib/libnxdk.lib
+
+CLEANRULES += clean-nxdk
+
+.PHONY: clean-nxdk
+clean-nxdk:
+	$(VE)rm -f $(NXDK_OBJS) $(NXDK_DIR)/lib/libnxdk.lib
+
+NXDK_AUTOMOUNT_D_SRCS := \
+	$(NXDK_DIR)/lib/nxdk/automount_d.c
+
+NXDK_AUTOMOUNT_D_OBJS = $(addsuffix .obj, $(basename $(NXDK_AUTOMOUNT_D_SRCS)))
+
+$(NXDK_DIR)/lib/libnxdk_automount_d.lib: $(NXDK_AUTOMOUNT_D_OBJS)
+
+ifneq ($(NXDK_DISABLE_AUTOMOUNT_D),y)
```

Comment:
nit: We should have a sample for this; if we don't add one in this PR, we'll need to create an issue after merge (otherwise this feature will be forgotten about).

The sample (or issue) should also explain that D: might still be mounted by the kernel (from [LaunchDataPage->launch_path](https://xboxdevwiki.net/Kernel/LaunchDataPage)); the Discord search doesn't work right now - but we discussed this a while ago (at least RadWolfie, LukeUsher and myself were involved). I believe I concluded that the `launch_path` was like `[<DVD path>;]<XBE path>` - but nobody tested or documented it. I don't fully remember if it supports relative paths for DVDs / absolute paths for XBE.